### ISSUE TITLE
Cannot generate sequence values when selecting from a distributed table despite push down to worker

### ISSUE THREAD

**Issue by kaikuchn** [citusdata/citus#2365]
Greetings,

I've been trying to insert values into a distributed table based on a select from a colocated distributed table. The table I'm inserting into has a serial id column and inserts are refused with the following error:

```
ERROR:  INSERT ... SELECT cannot generate sequence values when selecting from a distributed table
```

This is surprising since I thought all SQL features are supported when a query is pushed down to a single worker. Here's a simple demonstration:

```
=# create table foo (id serial, val text, tenant uuid);
=# select create_distributed_table('foo', 'tenant');
=# insert into foo values ("hey", uuid_generate_v1mc());
=# select * from foo;
 id | val |                tenant
----+-----+--------------------------------------
  1 | hey | 2c7268c4-b0e9-11e8-9439-af0c3c91f5b3
(1 row)
=# insert into foo (val, tenant) select val, tenant from foo where tenant = '2c7268c4-b0e9-11e8-9439-af0c3c91f5b3';
ERROR:  INSERT ... SELECT cannot generate sequence values when selecting from a distributed table
```

Trying to use CTEs will result in a less obvious error:
```
=# WITH bar AS (select val, tenant FROM foo WHERE tenant = '2c7268c4-b0e9-11e8-9439-af0c3c91f5b3') INSERT INTO foo (val, tenant) SELECT val, tenant FROM bar;
WARNING:  relation "public.foo_id_seq" does not exist
WARNING:  relation "public.foo_id_seq" does not exist
ERROR:  could not receive query results
```

For now I'm using `CREATE TABLE ... AS ...` as a work around. However, it'd be nice if I wouldn't have to jump through those hoops. And I also think that in case of the CTE some check is missing?

Best regards,
Kai

PS: I'm on a self-hosted Citus 7.5-7 with Postgres 10.5
```
=# SELECT * FROM run_command_on_workers($$SELECT e.extversion FROM pg_catalog.pg_extension e WHERE e.extname = 'citus'$$);
   nodename   | nodeport | success | result
--------------+----------+---------+--------
 4e0277051902 |     5432 | t       | 7.5-7
 84db803724d6 |     5432 | t       | 7.5-7
(2 rows)

=# SELECT version();
                                                             version
----------------------------------------------------------------------------------------------------------------------------------
PostgreSQL 10.5 (Debian 10.5-1.pgdg90+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 6.3.0-18+deb9u1) 6.3.0 20170516, 64-bit
```

**Comment by saicitus** [comment]
Customer has run into a similar issue with some differences: 

INSERT INTO SELECT was between the same table (source and destination was same) but the distribution column was being changed in the INSERT INTO SELECT. The goal for the customer was to de-duplicate data with a different sharding column and obviously increasing id sequence.

```
create table a_seq(id serial, dist_key int);
SELECT create_distributed_table('a_seq','dist_key');
INSERT INTO a_seq(dist_key) values (1)
INSERT INTO a_seq(dist_key) select dist_key*100 from a_seq;
ERROR:  INSERT ... SELECT cannot generate sequence values when selecting from a distributed table
```

**Comment by marcocitus** [comment]
Fixed by #3680
